### PR TITLE
governance: vote

### PR DESCRIPTION
**Pull request by Fraccaman**
- Add vote transaction
- Add governance VP logic to handle the vote
- Improve CLI proposal query command to check the status of a proposal

Based on #954

### CODE REVIEW

**Comment by Fraccaman** [comment]
pls update wasm

**Comment by Fraccaman** [comment]
pls update wasm

**Comment by Fraccaman** [comment]
pls update wasm

**Comment by Fraccaman** [comment]
pls update wasm

**Comment by grarco** [comment]
pls update wasm

**Comment by Fraccaman** [comment]
pls publish wasm

**Comment by Fraccaman** [comment]
pls publish wasm

**Comment by Fraccaman** [comment]
pls publish wasm

**Comment by Fraccaman** [comment]
pls publish wasm

**Comment by Fraccaman** [comment]
pls publish wasm

File: shared/src/types/token.rs [hunk 1]
```diff
@@ -71,6 +71,13 @@ impl Amount {
     pub fn max() -> Self {
         Self { micro: u64::MAX }
     }
+
+    /// Create amount from Change
```
Comment (tzemanovic) [inline shared/src/types/token.rs:75]:
```suggestion
    /// Create amount from Change
    /// 
    /// # Panics
    ///
    /// Panics if the change is negative or overflows `u64`.
```

**Review by tzemanovic — APPROVED** [review]
LGTM, thanks!

**Comment by tzemanovic** [comment]
pls update wasm

File: apps/src/lib/client/rpc.rs [hunk 2]
```diff
@@ -1253,3 +1259,175 @@ pub async fn query_result(_ctx: Context, args: args::QueryResult) {
         }
     }
 }
+
+pub async fn compute_tally(
```
Comment (tzemanovic) [inline apps/src/lib/client/rpc.rs:1263]: the full tallying implementation is in the next PR #1042

**Comment by tzemanovic** [comment]
pls update wasm

**Pull request merged**